src/exo/frontend/pyparser.py

```diff
@@ -35,54 +38,106 @@ def str_to_mem(name):
     return getattr(sys.modules[__name__], name)
 
 
+@dataclass
+class SourceInfo:
```

Comment:
Why did you need a wrapper for `SrcInfo`?

Reply:
I decided to make a wrapper because before we just passed a getsrcinfo function into Parser, which I thought was confusing because then you don't know where the behavior of getsrcinfo is defined. I think it is neater if it is wrapped in an object with a defined get_src_info method that you can jump to.